site/en/blog/show-picker/index.md

```diff
@@ -0,0 +1,203 @@
+
+I'm happy to share that those days will be over soon as the web platform is
+introducing a canonical way to show a browser picker for `<input>` elements with
+these types: `"date"`, `"month"`, `"week"`, `"time"`, `"datetime-local"`,
+`"color"`, and `"file"`. It is also working for `<input>` elements with
+suggestions powered by `<datalist>` or `"autocomplete"` which we'll cover as
+well in this article.
+
+<figure class="w-figure">
+{% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/uh0U2YQnUMato21MzzbR.png", alt="Screenshots of browser pickers", width="800", height="217" %}
+  <figcaption class="w-figcaption">Browser date pickers in Chrome Desktop, Chrome Mobile, Safari Desktop, Safari Mobile, and Firefox Desktop (July 2021).</figcaption>
+</figure>
+
+{% Aside %}
+Unlike a third-party picker widget, a browser picker is familiar to the user,
+has great specific platform support, and is always maintained as part of the
+browser.
+{% endAside %}
+
+## How to show a picker {: #how-to }
+
+Calling `showPicker()` on an `<input>` element returns a promise that resolves
+when a browser picker is shown to the user. It must be called in response to a
+user gesture such as a touch gesture or mouse click; otherwise it will reject.
+For security reasons,it will also reject when it's called in a cross-origin
+iframe.
+
+A browser picker is shown when the `<input>` element is one of these types:
+`"date"`, `"month"`, `"week"`, `"time"`, `"datetime-local"`, `"color"`, and
+`"file"`.
+
+The example below shows you how to open a browser date picker.
+
+```html/9
+<input type="date">
+<button>Show the date picker</button>
+
+<script>
+  const button = document.querySelector("button");
+  const dateInput = document.querySelector("input");
+
+  button.addEventListener("click", async () => {
+    try {
+      await dateInput.showPicker();
+      // A date picker is shown.
+    } catch (error) {
+      // Use external library when this fails.
+    }
+  });
+</script>
+```
+
+A browser picker may also be shown if the `<input>` element has some suggestions
+powered by `<datalist>` or `"autocomplete"`.
+
+The example below shows you how to open a browser picker with `<datalist>`.
+
+```html/16
+<datalist id="ice-cream-flavors">
+  <option value="Chocolate"> </option>
+  <option value="Coconut"> </option>
+  <option value="Mint"> </option>
+  <option value="Strawberry"> </option>
+  <option value="Vanilla"> </option>
+</datalist>
+<input type="text" list="ice-cream-flavors">
+<button>Show the suggestions</button>
+
+<script>
+  const button = document.querySelector("button");
+  const iceCreamFlavorsInput = document.querySelector("input");
+
+  button.addEventListener("click", async () => {
+    try {
+      await iceCreamFlavorsInput.showPicker();
+      // A picker containing some ice cream flavors is shown.
+    } catch (error) {
+      // Use external library when this fails.
+    }
+  });
+</script>
+```
+
+## Demo
+
+A playground demo is available at
```

Comment:
```suggestion
A demo is available at
```

Reply:
Or are you trying to distinguish this from other types of demos?

Reply:
There were inline demos in the previous web.dev post. That's not the case anymore. Thanks for catching!